idaes/apps/grid_integration/forecaster.py

```diff
@@ -251,3 +299,412 @@ def _forecast(self, means, stds, hour, horizon, n_samples):
+        )
+
+        return da_forecast, rt_forecast
+
+    def forecast_real_time_prices(self, date, hour, bus, horizon, n_samples):
+
+        """
+        Forecast real-time market prices.
+
+        Arguments:
+            date: intended date of the forecasts
+
+            hour: intended hour of the forecasts
+
+            bus: intended bus of the forecasts
+
+            horizon: number of the time periods of the forecasts
+
+            n_samples: number of the samples
+
+        Returns:
+            dict: real-time price forecasts
+
+        """
+
+        return self._forecast(
+            historical_price_dict=self.historical_rt_prices,
+            market="real-time",
+            date=date,
+            hour=hour,
+            bus=bus,
+            horizon=horizon,
+            n_samples=n_samples,
+        )
+
+    def forecast_day_ahead_prices(self, date, hour, bus, horizon, n_samples):
+
+        """
+        Forecast day-ahead market prices.
+
+        Arguments:
+            date: intended date of the forecasts
+
+            hour: intended hour of the forecasts
+
+            bus: intended bus of the forecasts
+
+            horizon: number of the time periods of the forecasts
+
+            n_samples: number of the samples
+
+        Returns:
+            dict: day-ahead price forecasts
+
+        """
+
+        return self._forecast(
+            historical_price_dict=self.historical_da_prices,
+            market="day-ahead",
+            date=date,
+            hour=0,
+            bus=bus,
+            horizon=horizon,
+            n_samples=n_samples,
+        )
+
+    def _forecast(
+        self, historical_price_dict, market, date, hour, bus, horizon, n_samples
+    ):
+        """
+        Forecast energy market prices using historical prices.
+
+        Arguments:
+
+            historical_price_dict: the dictionary that holds the intended historical prices
+
+            market: the market that the price forecast is for, e.g., day-ahead
+
+            date: intended date of the forecasts
+
+            hour: intended hour of the forecasts
+
+            bus: intended bus of the forecasts
+
+            horizon: number of the time periods of the forecasts
+
+            n_samples: number of the samples
+
+        Returns:
+            dict: price forecasts
+
+        """
+
+        if bus not in historical_price_dict:
+            raise ForecastError(f"No {bus} {market} price available.")
+
+        historical_price_len = len(historical_price_dict[bus])
+        n_days = historical_price_len // 24
+
+        forecast = {}
+
+        for i in range(n_samples):
+            day_idx = n_days - (i % n_days) - 1
+
+            forecast[i] = [
+                historical_price_dict[bus][t % historical_price_len]
+                for t in range(day_idx * 24 + hour, day_idx * 24 + hour + horizon)
+            ]
+
+        return forecast
+
+    def fetch_hourly_stats_from_prescient(self, prescient_hourly_stats):
+
```

Comment:
This may be more difficult to test; it may require calling Prescient in the testing harness.

Reply:
With duck typing, a mock `prescient_hourly_stats` for testing should be pretty simple:

```
class MockPrescientHourlyStats:
    def __init__(self, bus_to_prices_dict) -> None:
        self.observed_bus_LMPs = bus_to_prices_dict

@pytest.mark.unit
def test_fetch_hourly_stats_from_prescient(historical_da_prices, historical_rt_prices):
    backcaster = Backcaster(historical_da_prices, historical_rt_prices)
    prescient_hourly_stats = MockPrescientHourlyStats({"test_bus": [100] * 24 + [200] * 24 + [300] * 24})
    backcaster.fetch_hourly_stats_from_prescient(prescient_hourly_stats)
```